zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -0,0 +1,750 @@
+            q_step,
+            add_gadget: configure_gadget!(),
+            bitwise_gadget: configure_gadget!(),
+            byte_gadget: configure_gadget!(),
+            comparator_gadget: configure_gadget!(),
+            dup_gadget: configure_gadget!(),
+            error_oog_pure_memory_gadget: configure_gadget!(),
+            jump_gadget: configure_gadget!(),
+            jumpdest_gadget: configure_gadget!(),
+            jumpi_gadget: configure_gadget!(),
+            memory_gadget: configure_gadget!(),
+            pc_gadget: configure_gadget!(),
+            pop_gadget: configure_gadget!(),
+            push_gadget: configure_gadget!(),
+            signextend_gadget: configure_gadget!(),
+            stop_gadget: configure_gadget!(),
+            swap_gadget: configure_gadget!(),
+            step: step_curr,
+            presets_map,
+        };
+
+        Self::configure_lookup(
+            meta,
+            q_step,
+            fixed_table,
+            tx_table,
+            rw_table,
+            bytecode_table,
+            independent_lookups,
+        );
+
+        config
+    }
+
+    fn configure_gadget<G: ExecutionGadget<F>>(
+        meta: &mut ConstraintSystem<F>,
+        q_step: Selector,
+        randomness: &Expression<F>,
+        step_curr: &Step<F>,
+        step_next: &Step<F>,
+        independent_lookups: &mut Vec<Vec<Lookup<F>>>,
+        presets_map: &mut HashMap<ExecutionState, Vec<Preset<F>>>,
+    ) -> G {
+        let mut cb = ConstraintBuilder::new(
+            step_curr,
+            step_next,
+            randomness.clone(),
+            G::EXECUTION_STATE,
+        );
+
+        let gadget = G::configure(&mut cb);
+
+        let (constraints, lookups, presets) = cb.build();
+        assert!(
+            presets_map.insert(G::EXECUTION_STATE, presets).is_none(),
+            "execution state already configured"
+        );
+
+        if !constraints.is_empty() {
+            meta.create_gate(G::NAME, |meta| {
+                let q_step = meta.query_selector(q_step);
+
+                constraints.into_iter().map(move |(name, constraint)| {
+                    (name, q_step.clone() * constraint)
+                })
+            });
+        }
+
+        // Push lookups of this ExecutionState to independent_lookups for
+        // further configuration in configure_lookup.
+        independent_lookups.push(lookups);
+
+        gadget
+    }
+
+    fn configure_lookup<TxTable, RwTable, BytecodeTable>(
+        meta: &mut ConstraintSystem<F>,
+        q_step: Selector,
+        fixed_table: [Column<Fixed>; 4],
+        tx_table: TxTable,
+        rw_table: RwTable,
+        bytecode_table: BytecodeTable,
+        independent_lookups: Vec<Vec<Lookup<F>>>,
+    ) where
+        TxTable: LookupTable<F, 4>,
+        RwTable: LookupTable<F, 8>,
+        BytecodeTable: LookupTable<F, 4>,
+    {
+        // Because we constraint one and only one ExecutionState will be enabled
```

Comment:
```suggestion
        // Because one and only one ExecutionState is enabled
```

Reply:
Fixed in [`0b68140`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/0b681404db2130a86297f2fb325fd69ba3101501)